### PR TITLE
feat: migrate to MUI v4

### LINKED ISSUE #2013: Support the Material-UI version 4.x.x
- [ ] I have searched this repository's [issues](https://github.com/devexpress/devextreme-reactive/issues) and believe that this is not a duplicate.

### I'm using ...

- [x ] React Grid
- [ ] React Chart
- [ ] React Scheduler

### Current Behaviour

When I updated material-UI to version 4, there were some changes in spacing,
However, it seems that the library has not changed.

> Warning: Material-UI: theme.spacing.unit usage has been deprecated.
> It will be removed in v5.
> You can replace `theme.spacing.unit * y` with `theme.spacing(y)`.
in dx-react-grid-material-ui.es.js

### Screenshots

Screen Shot 2019-05-08 at 9 26 13 AM
Screen Shot 2019-05-08 at 9 25 56 AM

### Environment

- material-ui: ^4.0.0-beta.1

### PR DESCRIPTION
**Pull request by MaximKudriavtsev**
Fixes #2013

### CODE REVIEW

**Comment by MaximKudriavtsev** [comment]
We are waiting to fix this [bug](https://github.com/mui-org/material-ui/issues/15922).

**Comment by gunn4r** [comment]
Yay looks like that [bug](https://github.com/mui-org/material-ui/issues/15922) is fixed!

**Comment by MaximKudriavtsev** [comment]
Hi @gunn4r,

You are right! We have resumed this work already.

File: packages/dx-react-grid-demos/src/demo-sources/grid-featured-data-editing/material-ui/demo.jsx [hunk 1]
```diff
@@ -35,9 +35,7 @@ import {
 
 const styles = theme => ({
   lookupEditCell: {
-    paddingTop: theme.spacing.unit * 0.875,
-    paddingRight: theme.spacing.unit,
-    paddingLeft: theme.spacing.unit,
+    padding: theme.spacing(1),
```
Comment (DmitryBogomolov) [inline packages/dx-react-grid-demos/src/demo-sources/grid-featured-data-editing/material-ui/demo.jsx]: Are `paddingTop` and `paddingBottom` set to `1` here?

Comment (MaximKudriavtsev) [reply packages/dx-react-grid-demos/src/demo-sources/grid-featured-data-editing/material-ui/demo.jsx]: On this case, we have removed these paddings to make our markup consistency with previous builds.

**Comment by churkin** [comment]
Good job!

File: packages/dx-react-grid-material-ui/src/templates/column-chooser/item.jsx [hunk 2]
```diff
@@ -9,6 +9,9 @@ const styles = ({
   checkbox: {
     padding: 0,
   },
+  itemText: {
+    paddingLeft: 8,
```
Comment (Krijovnick) [inline packages/dx-react-grid-material-ui/src/templates/column-chooser/item.jsx]: Why you don't use `theme.spacing(1)` here?

**Pull request merged**